libr/core/cmd_flag.c

```diff
@@ -682,6 +682,78 @@ static int flag_space_stack_list(RFlag *f, int mode) {
 	return i;
 }
 
+typedef struct {
+	int rad;
+	PJ *pj;
+	RAnalFunction *fcn;
+} PrintFcnLabelsCtx;
+
+static bool print_function_labels_cb(void *user, const ut64 addr, const void *v) {
+	const PrintFcnLabelsCtx *ctx = user;
+	const char *name = v;
+	switch (ctx->rad) {
+	case '*':
+	case 1:
+		r_cons_printf ("f.%s@0x%08"PFMT64x"\n", name, addr);
+		break;
+	case 'j':
+		pj_kn (ctx->pj, name, addr);
+		break;
+	default:
+		r_cons_printf ("0x%08"PFMT64x" %s   [%s + %"PFMT64d"]\n",
+			addr,
+			name, ctx->fcn->name,
+			addr - ctx->fcn->addr, addr);
+	}
+	return true;
+}
+
+
+static void print_function_labels_for(RAnalFunction *fcn, int rad, PJ *pj) {
+	r_return_if_fail (fcn && (rad != 'j' || pj));
+	if (rad == 'j') {
+		pj_o (pj);
+	}
+	PrintFcnLabelsCtx ctx = { rad, pj, fcn };
+	ht_up_foreach (fcn->labels, print_function_labels_cb, &ctx);
+	if (rad == 'j') {
+		pj_end (pj);
+	}
+}
+
+static void print_function_labels(RAnal *anal, RAnalFunction *fcn, int rad) {
+	r_return_if_fail (anal || fcn);
```

Comment:
&& not ||

Reply:
No, || is correct. It either expects `fcn` in which case it prints only this function and ignores `anal` or `anal` in which case it prints all functions and ignores `fcn`.

Reply:
thats quite weird, and fcn have a pointer to anal. but its an internal api, so its probably not tied to clear usage arguments.

Reply:
also, anal is never used in this function

Reply:
Yep, the main reason to do this and not multiple functions is to avoid code duplication.

Reply:
anal is used here: https://github.com/radareorg/radare2/pull/17621/files#diff-1c86fab8e892153e8e316d8fc8ce810eR740

Reply:
yes i understand the code dup reason, but imho it would be better to specify what do in an argument or function name rather than doing black magic depending on which argument is null